ors-api/src/main/java/org/heigit/ors/api/services/AbstractApiService.java

```diff
@@ -0,0 +1,424 @@
+import org.heigit.ors.api.requests.routing.RouteRequest;
+import org.heigit.ors.common.DistanceUnit;
+import org.heigit.ors.common.StatusCode;
+import org.heigit.ors.exceptions.*;
+import org.heigit.ors.geojson.GeometryJSON;
+import org.heigit.ors.routing.*;
+import org.heigit.ors.routing.graphhopper.extensions.HeavyVehicleAttributes;
+import org.heigit.ors.routing.graphhopper.extensions.VehicleLoadCharacteristicsFlags;
+import org.heigit.ors.routing.graphhopper.extensions.WheelchairTypesEncoder;
+import org.heigit.ors.routing.graphhopper.extensions.reader.borders.CountryBordersReader;
+import org.heigit.ors.routing.parameters.ProfileParameters;
+import org.heigit.ors.routing.parameters.VehicleParameters;
+import org.heigit.ors.routing.parameters.WheelchairParameters;
+import org.heigit.ors.routing.pathprocessors.BordersExtractor;
+import org.heigit.ors.util.DistanceUnitUtil;
+import org.heigit.ors.util.GeomUtility;
+import org.json.simple.JSONObject;
+import org.locationtech.jts.geom.Geometry;
+import org.locationtech.jts.geom.MultiPolygon;
+import org.locationtech.jts.geom.Polygon;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+public abstract class AbstractApiService {
+
+    abstract EndpointsProperties getEndpointsProperties();
+    abstract double getMaximumAvoidPolygonArea();
+    abstract double getMaximumAvoidPolygonExtent();
+
+    public static String[] convertAPIEnumListToStrings(Enum[] valuesIn) {
+        String[] attributes = new String[valuesIn.length];
+
+        for (int i = 0; i < valuesIn.length; i++) {
+            attributes[i] = convertAPIEnum(valuesIn[i]);
+        }
+
+        return attributes;
+    }
+
+    protected static String convertAPIEnum(Enum valuesIn) {
+        return valuesIn.toString();
+    }
+
+    public static int convertVehicleType(APIEnums.VehicleType vehicleTypeIn, int profileType) throws IncompatibleParameterException {
+        if (!RoutingProfileType.isHeavyVehicle(profileType)) {
+            throw new IncompatibleParameterException(GenericErrorCodes.INVALID_PARAMETER_VALUE,
+                    "vehicle_type", vehicleTypeIn.toString(),
+                    APIRequest.PARAM_PROFILE, RoutingProfileType.getName(profileType));
+        }
+
+        if (vehicleTypeIn == null) {
+            return HeavyVehicleAttributes.UNKNOWN;
+        }
+
+        return HeavyVehicleAttributes.getFromString(vehicleTypeIn.toString());
+    }
+
+    protected static BordersExtractor.Avoid convertAvoidBorders(APIEnums.AvoidBorders avoidBorders) {
+        if (avoidBorders != null) {
+            switch (avoidBorders) {
+                case ALL:
+                    return BordersExtractor.Avoid.ALL;
+                case CONTROLLED:
+                    return BordersExtractor.Avoid.CONTROLLED;
+                default:
+                    return BordersExtractor.Avoid.NONE;
+            }
+        }
+        return null;
+    }
+
```

Comment:
Although most of the added code is in fact just code moved from the original classes, I've noticed that in this process some double blank lines have been introduced to separate method bodies. Was this change intentional?

Reply:
No, I removed many multiple blank lines, but maybe not all... I will reformat all changed classes.

Reply:
Removed some double blank lines, but decided not to reformat the classes NOW - Amandus mentioned, that a big reformatting of the whole project in an extra branch is planned. BTW in our .editorconfig we seem to allow double blank lines if I'm not wrong?